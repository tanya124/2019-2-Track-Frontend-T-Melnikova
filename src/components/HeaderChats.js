import React from 'react';
import { Link } from 'react-router-dom';
import styled from '@emotion/styled';
import '../styles/Header.css';
import SearchButton from './Buttons/SearchButton';
import BurgerButton from './Buttons/BurgerButton';
import Portal from './Portal';

const TopBar = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  background-color: blueviolet;
  height: 8vh;
`;

const Title = styled.span`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  margin-left: 5%;
  color: white;
  font-weight: bold;
  font-size: 2.5vh;
`;

const MenuListContainer = styled.div`
  position: absolute;
  top: 9vh;
  left: 10px;
  transform: translate(-10%, -10%);
  background-color: #f1f1f1;
  width: 130px;
  display: flex;
  flex-direction: column;
  padding: 5px;
  border-width: 0;
`;

const Button = styled.div`
  color: black;
  padding: 5px;
  border-bottom: 1px solid gray;
`;

function MenuList() {
	return (
		<Portal>
			<MenuListContainer>
				<Link className="linkChat" to='/2019-2-Track-Frontend-T-Melnikova/profile'>
					<Button>
					Редактировать профиль
					</Button>
				</Link>
			</MenuListContainer>
		</Portal>
	);
}

class HeaderChats extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			menuIsOpen: false,
		};

		this.changeStateMenu = this.changeStateMenu.bind(this);
	}

	changeStateMenu() {
		const current = this.state.menuIsOpen;
		this.setState({
			menuIsOpen: !current,
		});
	}

	render() {
		return (
			<TopBar>
				<BurgerButton changeStateMenu={this.changeStateMenu}/>
				{this.state.menuIsOpen && <MenuList />}
				<Title>Чаты</Title>
				<SearchButton />
			</TopBar>
		);
	}
}

export default HeaderChats;
