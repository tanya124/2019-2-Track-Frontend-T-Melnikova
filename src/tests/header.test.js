import React from 'react';
import renderer from 'react-test-renderer';
import HeaderChats from '../components/HeaderChats';

it('renders correctly', () => {
	const tree = renderer.create(<HeaderChats />).toJSON();
	expect(tree).toMatchSnapshot();
});
