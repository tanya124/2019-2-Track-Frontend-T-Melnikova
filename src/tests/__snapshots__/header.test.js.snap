// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-3cjx5i"
>
  <img
    className="css-1v7dot1"
    onClick={[Function]}
    src="menu.svg"
  />
  <span
    className="css-rgtn5"
  >
    Чаты
  </span>
  <img
    className="css-6jpwhn"
    src="search.svg"
  />
</div>
`;
